src/index.js

```diff
@@ -1,66 +1,211 @@
+      return null;
+    }
+    const user = await response.json();
+    return user.name || null;
+  } catch {
+    return null;
+  }
+}
+getUserNameAsync(1).then((name) => console.log(name));
+
+// 5
+function getAllPostsFromUser(userId) {
+  return new Promise((resolve, reject) => {
+    fetch('https://jsonplaceholder.typicode.com/posts')
+      .then((response) => response.json())
+      .then((posts) => {
+        const userPosts = posts.filter((post) => post.userId === userId);
+        if (userPosts.length > 0) {
+          resolve(userPosts);
         } else {
-          const firstInputValue = firstInput.value;
-          const secondInputValue = secondInput.value;
-          const inputsMatch = firstInputValue === secondInputValue;
-
-          if (
-            inputsMatch &&
-            secondInputValue !== '' &&
-            firstInputValue !== ''
-          ) {
-            resolve();
-          }
+          reject(new Error(`No posts found for user with ID ${userId}`));
         }
-      });
+      })
+      .catch((error) => reject(error));
+  });
+}
+
+getAllPostsFromUser(5)
+  .then((posts) => console.log(posts))
+  .catch((error) => console.error(error));
+
+async function getAllPostsFromUserAsync(userId) {
+  try {
+    const response = await fetch('https://jsonplaceholder.typicode.com/posts');
+    const posts = await response.json();
+    const userPosts = posts.filter((post) => post.userId === userId);
+    if (userPosts.length > 0) {
+      return userPosts;
+    } else {
+      throw new Error(`No posts found for user with ID ${userId}`);
+    }
+  } catch (error) {
+    throw error;
+  }
+}
+
+getAllPostsFromUserAsync(1)
+  .then((posts) => console.log(posts))
+  .catch((error) => console.error(error));
+
+// 6
+function getAlbumsWithPhotos() {
+  return Promise.all([
+    fetch('https://jsonplaceholder.typicode.com/albums').then((response) =>
+      response.json(),
+    ),
+    fetch('https://jsonplaceholder.typicode.com/photos').then((response) =>
+      response.json(),
+    ),
+  ]).then(([albums, photos]) => {
+    return albums.map((album) => {
+      album.photos = photos.filter((photo) => photo.albumId === album.id);
+      return album;
     });
   });
 }
 
-waitForPasswordsToMatch('#password-inputss', '#confirm-password-input')
-  .then(function () {
-    console.log('Passwords match');
-  })
-  .catch(function () {
-    console.log('Passwords doesn`t match or are missing');
+getAlbumsWithPhotos().then((albumsWithPhotos) => {
+  console.log(albumsWithPhotos);
+});
+
+async function getAlbumsWithPhotosAsync() {
+  const [albumsResponse, photosResponse] = await Promise.all([
+    fetch('https://jsonplaceholder.typicode.com/albums'),
+    fetch('https://jsonplaceholder.typicode.com/photos'),
+  ]);
+
+  const albums = await albumsResponse.json();
+  const photos = await photosResponse.json();
+
+  return albums.map((album) => {
+    album.photos = photos.filter((photo) => photo.albumId === album.id);
```

Comment:
This line runs 500000 times. That's a lot. It has the potential to be very far from optimal. Do you remember why? Do you remember how to fix it?